### PR TITLE
fixes radial menu gc issues

### LINKED ISSUE #19907: Radial menus fail to GC always
**Issue Description**:
image

Had dels bad.

I think its something to do with 
```

/obj/screen/radial
	icon = 'icons/mob/radial.dmi'
	layer = ABOVE_HUD_LAYER
	plane = ABOVE_HUD_PLANE
	var/datum/radial_menu/parent
	
``` 
never clearing the `parent` var

### PR DESCRIPTION
**Pull request by GDNgit**
## What Does This PR Do
fixes #19907 
cleaning up elements and the close button is needed
## Changelog
NPFC

### CODE REVIEW

**Review by S34NW — APPROVED** [review]

File: code/_onclick/hud/radial.dm [hunk 1]
```diff
@@ -254,6 +254,9 @@ GLOBAL_LIST_EMPTY(radial_menus)
 
 /datum/radial_menu/proc/hide()
 	if(current_user)
+		for(var/obj/screen/radial/slice/slice_to_remove in elements)
+			slice_to_remove.parent = null
+		close_button.parent = null
```
Comment (farie82) [inline code/_onclick/hud/radial.dm:259]: Shouldn't this be part of the Destroy instead? Also the elements themselves are not being deleted

Comment (GDNgit) [reply code/_onclick/hud/radial.dm]: yeah probably
also bwuh

File: code/_onclick/hud/radial.dm [hunk 2]
```diff
@@ -273,6 +270,12 @@ GLOBAL_LIST_EMPTY(radial_menus)
 /datum/radial_menu/Destroy()
 	Reset()
 	hide()
+	for(var/obj/screen/radial/slice/slice_to_remove in elements)
+		slice_to_remove.parent = null
```
Comment (farie82) [inline code/_onclick/hud/radial.dm:274]: This should be done in the `Destroy` of `/obj/screen/radial`. This also allows you to use `QDEL_LIST_CONTENTS`

File: code/_onclick/hud/radial.dm [hunk 3]
```diff
@@ -273,6 +270,12 @@ GLOBAL_LIST_EMPTY(radial_menus)
...
+		qdel(slice_to_remove)
+	close_button.parent = null
```
Comment (farie82) [inline code/_onclick/hud/radial.dm:276]: Also allows you to remove this one

File: code/_onclick/hud/radial.dm [hunk 4]
```diff
@@ -270,6 +270,12 @@ GLOBAL_LIST_EMPTY(radial_menus)
...
+	qdel(close_button)
```
Comment (farie82) [inline code/_onclick/hud/radial.dm:277]:
```suggestion
	QDEL_NULL(close_button)
```

File: code/_onclick/hud/radial.dm (hunk 2 above)
Comment (GDNgit) [reply code/_onclick/hud/radial.dm:274]: Good idea, I'll implement it when I get onto my PC (eta 2 hours)

Comment (GDNgit) [reply code/_onclick/hud/radial.dm]: Done

**Review by farie82 — APPROVED** [review]

**Pull request merged**